pkg/controller/machine/controller.go

```diff
@@ -217,8 +219,13 @@ func (r *ReconcileMachine) Reconcile(request reconcile.Request) (reconcile.Resul
 		}
 
 		if err := r.actuator.Delete(ctx, cluster, m); err != nil {
-			klog.Errorf("Failed to delete machine %q: %v", name, err)
-			return delayIfRequeueAfterError(err)
+			// if we have an invalid zone/project error and there are no addresses,
+			// we can continue; otherwise we need to return an error and
+			// requeue.
+			if !isInvalidMachineConfiguration(err) || len(m.Status.Addresses) > 0 {
```

Comment:
`isInvalidMachineConfiguration` will take care of the case where the configuration is invalid from the beginning. `len(m.Status.Addresses) > 0` will handle the case when a machine configuration was invalidated after an instance was created. So only a small window is left when we can lose instances, e.g. right after request to create one was sent and before a list of node addresses was set. I would have to run the actuators to see how much time it takes to set the node addresses. Though, Azure will be the slowest, Kubemark the fastest.

Question is what's the likelihood of machineset getting scaled up and then scaled down after a request to create an instance was sent and before the list of addresses was set?

Reply:
`s/isInvalidMachineConfiguration/isValidMachineConfiguration` instead of using double negation.

Reply:
> We could call it isInvalidMachineConfigurationError but that's redundant based on the input parameters IMO.

I think that would be an improvement and quite idiomatic.

Reply:
> `isInvalidMachineConfiguration` will take care of the case where the configuration is invalid from the beginning. `len(m.Status.Addresses) > 0` will handle the case when a machine configuration was invalidated after an instance was created. So only a small window is left when we can loose instances, e.g. right after request to create one was sent and before a list of node addresses was set.

I think we should add your text as the comment to explain the intent and/or rationale. /cc @michaelgugino